File: 1208. Get Equal Substrings Within Budget.go
func equalSubstring(s string, t string, maxCost int) int {
    left, current, answer := 0, 0, 0
    for right := 0; right < len(s); right++ {
        current += abs(int(s[right]) - int(t[right]))
        for left <= right && current > maxCost {
            current -= abs(int(s[left]) - int(t[left]))
            left++
        }
        answer = max(answer, right - left + 1)
    }
    return answer
}

func abs(i int) int {
    if i < 0 {
        return -i
    }
    return i
}

File: 1208. Get Equal Substrings Within Budget.java
class Solution {
    public int equalSubstring(String s, String t, int maxCost) {
        int left = 0;
        int current = 0;
        int answer = 0;
        for (int right = 0; right < s.length(); right++) {
            current += Math.abs(s.charAt(right) - t.charAt(right));
            while (left <= right && current > maxCost) {
                current -= Math.abs(s.charAt(left) - t.charAt(left));
                left++;
            }
            answer = Math.max(answer, right - left + 1);
        }
        return answer;
    }
}
